File: models/claims/medication.js
const { DataTypes } = require('sequelize');
const sequelize = require('../../config/database');

const Medicine = sequelize.define('Medicine', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  code: {
    type: DataTypes.STRING,
    unique: true,
  },
  generic_name: DataTypes.STRING,
  unit_of_pricing: {
    type: DataTypes.STRING,
    allowNull: false,
    defaultValue: 'per unit',
  },
  market_price: {
    type: DataTypes.FLOAT,
    allowNull: true,
    defaultValue: 0
  },
  nhia_price:{
    type: DataTypes.FLOAT,
    allowNull: false,
    defaultValue: 0
  },
  is_nhia_covered:{
    type:DataTypes.BOOLEAN,
    defaultValue:true,
  },
  level_of_prescribing: DataTypes.STRING,
}, {
  tableName: 'medicines',
  timestamps: false
});

Medicine.associate = (models) => {
  Medicine.hasMany(models.institutionPharmacyPrice, {
    foreignKey: 'medicine_id',
    as: 'institutionPharmacyPrices'
  });
};

module.exports = Medicine;

File: claims-service/app/models/medication.py
"""Medication model - NHIA medication catalog."""

import uuid
from datetime import datetime

from sqlalchemy import Column, DateTime, Float, String, Boolean
from sqlalchemy.dialects.postgresql import UUID

from app.database import Base


class Medication(Base):
    __tablename__ = "medicines"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    code = Column(String(50), unique=True, nullable=True, index=True)
    generic_name = Column(String(300), nullable=True)
    unit_of_pricing = Column(String(50), nullable=False, default="per unit")
    market_price = Column(Float, default=0)
    nhia_price = Column(Float, default=0)
    is_nhia_covered = Column(Boolean, default=True)
    level_of_prescribing = Column(String(50), nullable=True)

    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
    updated_at = Column(DateTime(timezone=True), default=datetime.utcnow, onupdate=datetime.utcnow)
